inc/class_feedgeneration.php

```diff
@@ -121,17 +135,42 @@ function generate_feed()
 			}
 			switch($this->feed_format)
 			{
-				// Output Atom 1.0 format feed.
+				// Output JSON formatted feed.
+				case "json":
+					++$serial;
+					$end = $serial < count($this->items) ? "," : "";
+					$item_id = explode('tid=', $item['link']);
```

Comment:
Wouldn't it be easier to just build an army for the entry, then `json_encode` the whole array? eg:

```php
$entry = array(
	'url' => $item['link'],
	'title' => $item['title'],
	...
);

if(!empty($item['author']))
{
	$entry['author'] = array(
		...
	);
}

$this->feed .= json_encode($entry);
$this->feed .= $end . "\n";
```

The same would apply to the feed header. If nothing else, it's probably easier to read with an array in my opinion.

Reply:
I understand what you say. The reason doing it that way is to maintain the indentation in the similar manner and inline with how it has been done already for xml feeds.
If you don't bother about manual beautifying and insist I can do the suggested change. Please confirm.